systemtest/sourcemap_test.go

```diff
@@ -0,0 +1,349 @@
+	uploadSourcemap(t, srv, "../testdata/sourcemap/bundle.js.map",
+		"http://localhost:8000/test/e2e/general-usecase/bundle.js.map",
+		"apm-agent-js",
+		"2.0",
+	)
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-sourcemap", nil)
+
+	systemtest.SendRUMEventsPayload(t, srv, "../testdata/intake-v2/transactions_spans_rum_2.ndjson")
+	result := systemtest.Elasticsearch.ExpectDocs(t, "apm-*-span", nil)
+
+	systemtest.ApproveEvents(
+		t, t.Name(), result.Hits.Hits,
+		// RUM timestamps are set by the server based on the time the payload is received.
+		"@timestamp", "timestamp.us",
+	)
+}
+
+func TestFetchLatestSourcemap(t *testing.T) {
+	systemtest.CleanupElasticsearch(t)
+	srv := apmservertest.NewUnstartedServer(t)
+	srv.Config.RUM = &apmservertest.RUMConfig{Enabled: true}
+	err := srv.Start()
+	require.NoError(t, err)
+
+	// upload sourcemap file that finds no matchings
+	uploadSourcemap(t, srv, "../testdata/sourcemap/bundle.js.map",
+		"http://localhost:8000/test/e2e/general-usecase/bundle.js.map",
+		"apm-agent-js",
+		"2.0",
+	)
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-sourcemap", nil)
+
+	systemtest.SendRUMEventsPayload(t, srv, "../testdata/intake-v2/errors_rum.ndjson")
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-error", nil)
+	expectLogMessage(t, srv, "No Sourcemap available")
+	deleteIndex(t, "apm-*-error*")
+
+	// upload second sourcemap file with same key,
+	// that actually leads to proper matchings
+	// this also tests that the cache gets invalidated,
+	// as otherwise the former sourcemap would be taken from the cache.
+	// upload sourcemap file that finds no matchings
+	uploadSourcemap(t, srv, "../testdata/sourcemap/bundle.js.map",
+		"http://localhost:8000/test/e2e/general-usecase/bundle.js.map",
+		"apm-agent-js",
+		"1.0.1",
+	)
+	systemtest.Elasticsearch.ExpectMinDocs(t, 2, "apm-*-sourcemap", nil)
+
+	systemtest.SendRUMEventsPayload(t, srv, "../testdata/intake-v2/errors_rum.ndjson")
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-error", nil)
+	expectSourcemapUpdated(t)
+}
+
+func TestSourcemapCacheUsage(t *testing.T) {
+	systemtest.CleanupElasticsearch(t)
+	srv := apmservertest.NewUnstartedServer(t)
+	srv.Config.RUM = &apmservertest.RUMConfig{Enabled: true}
+	err := srv.Start()
+	require.NoError(t, err)
+
+	// upload valid sourcemap
+	uploadSourcemap(t, srv, "../testdata/sourcemap/bundle.js.map",
+		"http://localhost:8000/test/e2e/general-usecase/bundle.js.map",
+		"apm-agent-js",
+		"1.0.1",
+	)
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-sourcemap", nil)
+
+	// trigger cache
+	systemtest.SendRUMEventsPayload(t, srv, "../testdata/intake-v2/errors_rum.ndjson")
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-error", nil)
+	expectSourcemapUpdated(t)
+
+	// delete sourcemap index
+	deleteIndex(t, "apm-*-sourcemap*")
+
+	// index error document again
+	systemtest.SendRUMEventsPayload(t, srv, "../testdata/intake-v2/errors_rum.ndjson")
+	systemtest.Elasticsearch.ExpectDocs(t, "apm-*-error", nil)
```

Comment:
we should either delete the old error, or use ExectMinDocs with >1 here, otherwise the check would be satisfied by the error indexed from the first SendRUMEventsPayload call won't it?

Reply:
right, missed that. thanks